scripts/validate_test_schemas.py

```diff
@@ -0,0 +1,137 @@
+import json
+import logging
+import os
+import re
+import subprocess
+import sys
+import time
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+def check_connection():
+    checks = 4
+    while checks > 0:
+        response = subprocess.run(
+            [
+                "curl",
+                "-so",
+                "/dev/null",
+                "-w",
+                "%{http_code}",
+                "http://localhost:5002/status",
+            ],
+            capture_output=True,
+            text=True,
+            check=False,
+        ).stdout.strip()
+
+        if response != "200":
+            logging.error("\033[31m---Error: Schema Validator Not Reachable---\033[0m")
+            logging.error("\033[31mHTTP Status: %s\033[0m", response)
+            if checks != 1:
+                logging.info("Retrying...\n")
+                time.sleep(5)
+            else:
+                logging.info("Exiting...\n")
+                sys.exit(1)
+            checks -= 1
+        else:
+            checks = 0
```

Comment:
Minor, but this could be made a bit simpler to remove the nesting e.g. something like
```suggestion
def check_connection():
    for connection_attempts in range(4, 0, -1):
        response = subprocess.run(
            [
                "curl",
                "-so",
                "/dev/null",
                "-w",
                "%{http_code}",
                "http://localhost:5002/status",
            ],
            capture_output=True,
            text=True,
            check=False,
        ).stdout.strip()

        if response == "200":
            return

        logging.error("\033[31m---Error: Schema Validator Not Reachable---\033[0m")
        logging.error("\033[31mHTTP Status: %s\033[0m", response)

        if connection_attempts == 1:
            logging.info("Exiting...\n")
            sys.exit(1)

        logging.info("Retrying...\n")
        time.sleep(5)
```